libflux/flux-core/src/semantic/nodes.rs

```diff
@@ -1742,13 +1754,19 @@ impl RegexpLit {
 pub struct StringLit {
     pub loc: ast::SourceLocation,
     pub value: String,
+    /// The (label) type if label types are enabled. Should be removed once label polymorphism are
+    /// stabilized
+    pub typ: Option<MonoType>,
```

Comment:
Why would this field be removed? It seems like a string literal can be typed as either a label or a string, so wouldn't this field need to remain? I must be missing something.

Reply:
`StringLit::type_of` could construct the type dynamically as `MonoType::Label(Label::from(&self.value))` instead of having it stored. But storing it may be preferable anyway, removed the comment.